### PR TITLE
Split AST source into multiple compile units

### PR DESCRIPTION
**Pull request by mgreter**
This splits the massive AST source files into

- nodes (base)
- containers
- blocks (has_blocks)
- expressions
- statements
- values
- selectors

This is only a first step and we should probably do the same for some other massive source files (parser, eval and function). This also helps to keep memory requirements lower during compiling, although I'm not sure if we loose any noticeable performance from weaker optimization (related to the discussion about unified builds). I haven't put too much thought into which includes to put where, was more interested to get it to compile on all platforms. From a code maintainance point this is definitely a step in the right direction, so if CI goes green I would like to ship this (//CC @xzyfer).

### CODE REVIEW

**Comment by mgreter** [comment]
Seems to not some work on the autotools side. Have to test localy.

**Comment by xzyfer** [comment]
I need time to look over this. Going to bed now. Will make time tomorrow.

**Comment by mgreter** [comment]
Note: with autotools 2.69 we get a new warning:
```
automake-1.15: warning: possible forward-incompatibility.
automake-1.15: At least a source file is in a subdirectory, but the 'subdir-objects'
automake-1.15: automake option hasn't been enabled.  For now, the corresponding output
automake-1.15: object file(s) will be placed in the top-level directory.  However,
automake-1.15: this behaviour will change in future Automake versions: they will
automake-1.15: unconditionally cause object files to be placed in the same subdirectory
automake-1.15: of the corresponding sources.
automake-1.15: You are advised to start using 'subdir-objects' option throughout your
automake-1.15: project, to avoid future incompatibilities.
```

This happens also on the old code. Unfortunately building sassc/tester fails if I enable `subdir-objects` also for the main makefile. For now I only enabled it on the src directory, since it is needed to build correctly for this patch as it makes sense to use a folder structure when spliting up big source chunks.

I also looked at another warning, which seems to be a bug with autotools and GNUMakefile.am:
http://www.archivum.info/bug-libtool@gnu.org/2008-11/00081/Re-libtool-version-2.2.6-only-checks-Makefile.am-for-ACLOCAL_AMFLAGS.html

File: src/ast/blocks.cpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#include "../sass.hpp"
+#include "values.hpp"
```
Comment (xzyfer) [inline src/ast/blocks.cpp]: Should this be `blocks.hpp`?

File: src/ast/expressions.cpp [hunk 2]
```diff
@@ -0,0 +1,139 @@
+#include "../sass.hpp"
+#include "values.hpp"
```
Comment (xzyfer) [inline src/ast/expressions.cpp]: ditto?

File: src/ast/nodes.cpp [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#include "../sass.hpp"
+#include "values.hpp"
```
Comment (xzyfer) [inline src/ast/nodes.cpp]: ditto

File: src/ast/statements.cpp [hunk 4]
```diff
@@ -0,0 +1,19 @@
+#include "../sass.hpp"
+#include "values.hpp"
```
Comment (xzyfer) [inline src/ast/statements.cpp]: ditto

**Comment by xzyfer** [comment]
All in all :+1:
Not sold on `containers` but we change easily change it later if it's annoying enough.

File: src/ast/nodes.hpp [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+  inline Statement::~Statement() { }
+}
+
+
+
+/////////////////////////////////////////////////////////////////////////////////////
+// Hash method specializations for std::unordered_map to work with Sass::Expression
+/////////////////////////////////////////////////////////////////////////////////////
+
+namespace std {
```
Comment (xzyfer) [inline src/ast/nodes.hpp]: I think this belongs in `containers`. It's critical to understanding `Hashed`. Not entire sure how that'll work given it's a fwd decl.

**Comment by mgreter** [comment]
As I said, haven't put much thought into the header includes.
For now I just made it work, so I agree this should be refined further.

**Comment by xzyfer** [comment]
Yep just leaving some thoughts here so that they're somewhere.

**Pull request closed without merging**